server/authz/authz.go

```diff
@@ -126,17 +126,26 @@ func (a *Authorizer) TeamAuthorize(ctx context.Context, teamID uint, action stri
 	}
 
 	// global admins and maintainers are authorized to work with teams
-	if subject.GlobalRole != nil && *subject.GlobalRole != fleet.RoleObserver {
-		return nil
+	if subject.GlobalRole != nil {
+		switch *subject.GlobalRole {
+		case fleet.RoleAdmin:
+			return nil
+		case fleet.RoleMaintainer:
```

Comment:
Minor nit, but you can put both options in the same `case`, separated with a comma.